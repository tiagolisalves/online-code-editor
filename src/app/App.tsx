import React, { useEffect, useState } from 'react'
import { SideBar } from './SideBar/SideBar'
import { ActionsBar } from './ActionsBar/ActionsBar'
import { Editor } from './Editor/Editor'

import { AppService } from './AppService'
import { IFile, IFileTree } from './Models'

function App() {
    const [fileTree, setFileTree] = useState([] as IFileTree[])
    const [file, setFile] = useState({} as IFile)
    const load = () => {
        AppService.getFileTree().then((fT) => {
            setFileTree(fT)
        })
    }

    useEffect(() => {
        load()
    }, [])

    const openFile = (fileId: number) => {
        AppService.getFile(fileId).then((fT) => {
            setFile(fT)
        })
    }

    const saveFile = (fileParam: IFile) => {
        return () => {
            AppService.updateFile(fileParam)
        }
    }

    const deleteFile = (fileParam: IFile) => {
        return () => {
            AppService.deleteFile(fileParam.id).then(() => {
                load()
            })
        }
    }

    const onEdit = (fileContentParam: string) => {
        file.content = fileContentParam
    }

    return (
        <div className="app-container">
            <main>
                <div className="side-bar-container">
                    <div className="side-bar-title">EXPLORER</div>
                    <div className="side-bar">{fileTree ? <SideBar fileTree={fileTree} openFile={openFile}></SideBar> : <></>}</div>
                </div>
                <div className="editor-container">
                    <ActionsBar deleteFile={deleteFile(file)} saveFile={saveFile(file)}>
                        {file.name}
                    </ActionsBar>
                    <Editor fileContent={file.content} onEdit={onEdit}></Editor>
                </div>
            </main>
        </div>
    )
}

export { App }
