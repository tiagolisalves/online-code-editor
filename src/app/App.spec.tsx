import { create } from 'react-test-renderer'
import React from 'react'
import { App } from './App'

test("It should render 'App'", () => {
    const comp = create(<App></App>)
    expect(comp.toJSON()).toMatchSnapshot()
})
