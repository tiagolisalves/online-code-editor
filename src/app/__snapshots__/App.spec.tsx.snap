// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render 'App' 1`] = `
<div
  className="app-container"
>
  <main>
    <div
      className="side-bar-container"
    >
      <div
        className="side-bar-title"
      >
        EXPLORER
      </div>
      <div
        className="side-bar"
      >
        <span>
          Loading...
        </span>
      </div>
    </div>
    <div
      className="editor-container"
    >
      <div />
      <div
        className="editor-area"
        contentEditable="true"
        onInput={[Function]}
        suppressContentEditableWarning={true}
      >
        <pre>
          <code />
        </pre>
      </div>
    </div>
  </main>
</div>
`;
